src/main/kotlin/com/serameracorp/pattern/PatternService.kt

```diff
@@ -0,0 +1,140 @@
+package com.serameracorp.pattern
+
+import com.serameracorp.project.Project
+import com.serameracorp.project.projectFromResultSet
+import io.ktor.http.*
+import io.ktor.server.plugins.*
+import java.sql.ResultSet
+
+class PatternService {
+
+    val patternRepository = PatternRepository()
+    fun createPattern(formParams: Parameters): Int {
+        val nameParam = formParams["name"]
+        val publisherParam = formParams["publisher"]
+        val publishedInParam = formParams["published_in"]
+        val difficultyParam = formParams["difficulty"]
```

Comment:
I'd expect this logic to live in PatternPage, and have the service not deal with HTTP parameter lists.